From b68ee87b69ad4f1e921b50bf5bc2be0f167ed45c Mon Sep 17 00:00:00 2001
From: Geetha sowjanya <gakula@marvell.com>
Date: Wed, 29 Apr 2020 17:24:19 +0530
Subject: [PATCH 546/767] octeontx2-pf: Fix TCP/UDP checksum offload for IPv6
 pkts

commit 6ff2f961f218e07b0104fbac44b76bcea5979733 from
git@git.assembla.com:cavium/WindRiver.linux.git

For TCP/UDP checksum offload HW expects L3TYPE to be set
irrespective of IP header checksum is being offloaded or
not. Currently for IPv6 pkts L3TYPE is not being set
resulting in pkts with checksum error. This patch fixes
this issue.

Change-Id: Ibb090fb6bc14d08e29ce4c6c5731759262c207fe
Signed-off-by: Geetha sowjanya <gakula@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/27674
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.c
index c3d4dcd2aefb..4113cde3dc78 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.c
@@ -642,6 +642,7 @@ static void otx2_sqe_add_hdr(struct otx2_nic *pfvf, struct otx2_snd_queue *sq,
 			sqe_hdr->ol3type = NIX_SENDL3TYPE_IP4_CKSUM;
 		} else if (skb->protocol == htons(ETH_P_IPV6)) {
 			proto = ipv6_hdr(skb)->nexthdr;
+			sqe_hdr->ol3type = NIX_SENDL3TYPE_IP6;
 		}
 
 		if (proto == IPPROTO_TCP)
-- 
2.31.1

